src/map.c

```diff
@@ -560,6 +560,68 @@ int wally_map_keypath_get_item_fingerprint(const struct wally_map *map_in, size_
     return wally_keypath_get_fingerprint(item->value, item->value_len, bytes_out, len);
 }
 
+int wally_keypath_get_path_len(const unsigned char *val, size_t val_len,
+                               size_t *written)
+{
+    if (written)
+        *written = 0;
+    if (!val || val_len < BIP32_KEY_FINGERPRINT_LEN || val_len % sizeof(uint32_t))
+        return WALLY_EINVAL;
+    if (val_len > BIP32_KEY_FINGERPRINT_LEN)
+        *written = (val_len / sizeof(uint32_t)) - 1;
```

Comment:
nit:  this code assumes that BIP32_KEY_FINGERPRINT_LEN == sizeof(uint32_t)  [which ofc it does].
But it might be better to not ?
eg:
`if (!val || val_len < BIP32_KEY_FINGERPRINT_LEN || (val_len - BIP32_KEY_FINGERPRINT_LEN) % sizeof(uint32_t))`
and:
`*written = (val_len - BIP32_KEY_FINGERPRINT_LEN) / sizeof(uint32_t));`
?

Reply:
Agreed, consistency is better. I've added two small helpers to do these calcs and changed the code to use them, thanks.